src/hotspot/share/opto/phaseX.cpp

```diff
@@ -469,6 +469,11 @@ PhaseRenumberLive::PhaseRenumberLive(PhaseGVN* gvn,
 
   uint worklist_size = worklist->size();
 
+  GrowableArray<Node_Notes*>* old_node_note_array = C->node_note_array();
+  if (old_node_note_array != NULL) {
+    C->set_node_note_array(new (C->comp_arena()) GrowableArray<Node_Notes*> (C->comp_arena(), 8, 0, NULL));
```

Comment:
Use `nullptr` in these lines.
Can you use `_useful.size()` as initial array length?

Reply:
Thanks for the review, Vladimir. I updated the `NULL` usages.

> Can you use _useful.size() as initial array length?

The `node_note_array` uses buckets/blocks of size `C->_node_notes_block_size == 256`. So the actual required size would be `ceil((double)useful.size() / (double)256)` but we could simply use `1 + (useful.size() / 256)`. 

Now even when initializing the GrowableArray to that size, setting notes will then still call `Compile::grow_node_notes` multiple times to create the `Node_Notes` buckets and since it always at least doubles the backing array size, we actually end up with an array that is larger than what is required:
https://github.com/openjdk/jdk/blob/4619e8bae838abd1f243c2c65a538806d226b8e8/src/hotspot/share/opto/compile.cpp#L1233-L1238

I updated the code to properly pre-size the structure by calling `grow_node_notes`. This also has the advantage that the `Node_Notes` arena is one big chunk instead of incrementally allocating small ones.

What do you think?